tests/modules/csm/test_csm_module.py

```diff
@@ -493,3 +498,249 @@ def test_collect_data_fulfilled_state(
     # assert that it is not early return from function
     msg = list(filter(lambda log: "All epochs are already processed. Nothing to collect" in log, caplog.messages))
     assert len(msg) == 0, "Unexpected message found in logs"
+
+
+@dataclass(frozen=True)
+class BuildReportTestParam:
+    prev_root: str
+    prev_cid: str
+    prev_acc_shares: Iterable[tuple[NodeOperatorId, int]]
+    curr_distribution: Mock
+    curr_root: str
+    curr_cid: str
+    curr_log: str
```

Comment:
We have two CIDs, tree and log. It's not very clear from this params.

Reply:
fixed